code/game/objects/items/stacks/tape.dm

```diff
@@ -15,6 +15,23 @@
 	splint_factor = 0.65
 	merge_type = /obj/item/stack/sticky_tape
 	var/list/conferred_embed = EMBED_HARMLESS
+	var/obj/tape_gag = /obj/item/clothing/mask/muzzle/tape //The tape type you get when ripping off a piece of tape.
+	greyscale_config = /datum/greyscale_config/tape
+	greyscale_colors = "#B2B2B2#BD6A62"
+
+/obj/item/stack/sticky_tape/attack_hand(mob/user, list/modifiers)
+	if(user.get_inactive_held_item() == src)
+		if(is_zero_amount(delete_if_zero = TRUE))
+			return
+		user.put_in_hands(new tape_gag)
```

Comment:
what if it fails

Reply:
did melbert's commit fix what you were talking about?

Reply:
I changed it to work similarly to how bluespace polycrystals do it. Should work with `put_in_hands`'s built system to drop the item on the floor if it fails.